### PR TITLE
handle .nvmrc

### PR DESCRIPTION
**Pull request by blob42**
Currently `nvm use` with .nvmrc is broken. This fixes it.

### CODE REVIEW

**Comment by brigand** [comment]
@sp4ke I don't see how this would work. bash runs, some env vars get updated, and then it exits. This doesn't affect the parent shell (the fish session) at all.

Supporting `.nvmrc` is a good idea, but we should do it by changing `brigand_nvm_fish_find_matching_version` to set `target` based on `.nvmrc`, or something like that. Want to take another stab at it?

**Comment by blob42** [comment]
Right I totally missed it. I updated the PR using the find function and tried to handle aliases on the way.

**Comment by brigand** [comment]
- use `$nvm_alias_folder/*` instead of `ls` because people often override `ls`
- fish code conventionally uses tabs, not spaces
- remove logging, this command should be silent except in fatal error cases
- you still have the `bash` command which does nothing useful

Thanks for working on this.

**Comment by blob42** [comment]
Thanks for the feedback. Let me know if this is better. I fixed some bugs I stumbled upon when using aliases

**Comment by brigand** [comment]
Thanks! I'll merge this and then fix any issues I find.

File: nvm.fish [hunk 1]
```diff
@@ -63,14 +96,29 @@ end
...
 	if test $command = 'use'
-		set -l target_version $argv[2]
+
+		set -l nvmrc './.nvmrc'
+		# Use .nvmrc through node
+		if test (count $argv) -eq 1
+			if test -e $nvmrc
+				set target_version (cat $nvmrc)
+			else
+				bash -c "source ~/.nvm/nvm.sh; nvm $argv"
```
Comment (blob42) [inline nvm.fish:114]: I forgot to remove this line

**Comment by brigand** [comment]
`grep -P` wasn't work on osx, so I changed it to `grep -E`. That should also work on linux.

For some reason, no matter what I do it's selecting 6.10.0. Could you see if the latest code works for various node versions and aliases?

**Comment by brigand** [comment]
I reverted the PR until this is resolved. We can work in `nvm.beta.fish` which is the new version.

**Pull request merged**